src/remote/remote.go

```diff
@@ -515,6 +515,41 @@ func (c *Client) downloadActionOutputs(ctx context.Context, ar *pb.ActionResult,
 	return nil
 }
 
+// DownloadInputs downloads all the inputs of a remotely built target into the target directory.
+func (c *Client) DownloadInputs(target *core.BuildTarget, targetDir string, isTest bool) error {
+	var pbDir *pb.Directory
+	// first ensure all inputs are in the CAS
+	err := c.uploadBlobs(func(ch chan<- *uploadinfo.Entry) error {
+		defer close(ch)
+
+		var err error
+		pbDir, err = c.uploadInputs(ch, target, isTest)
+		if err != nil {
+			return err
+		}
+
+		return nil
```

Comment:
You can just `return err` without the nil check. If it's nil then that's fine.

Reply:
eh, of course ...